hawkbit-repository/hawkbit-repository-api/src/main/java/org/eclipse/hawkbit/repository/ControllerManagement.java

```diff
@@ -429,4 +429,26 @@ Target updateControllerAttributes(@NotEmpty String controllerId, @NotNull Map<St
      */
     @PreAuthorize(SpringEvalExpressions.IS_CONTROLLER)
     Action cancelAction(long actionId);
+
+    /**
+     * Updates given {@link Action} with its external id.
+     * 
+     * @param actionId
+     *            to be updated
+     * @param externalRef
+     *            of the action
+     */
+    @PreAuthorize(SpringEvalExpressions.IS_CONTROLLER)
```

Comment:
Move these methods to DeploymentManagement, change the permission(refer to the exampels ind eployment management).